.azure-pipelines/generation-templates/capture-openapi.yml

```diff
@@ -13,55 +13,86 @@ parameters:
+  steps:
+  - pwsh: |
+      $dirPath = "./conversion-settings"
+      $endpoint = "${{ parameters.endpoint }}"
+      Write-Host "endpoint: $endpoint"
+      Write-Host "directory path: $dirPath"
+      $jsonBase = @{}
+      foreach ($f in Get-ChildItem $dirPath -Recurse -Include *.json) {
+        $name = $f.Name.replace('.json','')
+        $data = @{"File"=$f.Name;"Name"=$name;}
+        $key = "$endpoint-$name"
+        $jsonBase.Add($key,$data)
+      }
+      $json = $jsonBase | ConvertTo-Json -Compress
+      Write-Host "##vso[task.setvariable variable=targets;isOutput=true]$json"
+    name: setTargets
 
-- template: checkout-metadata.yml
-- template: set-user-config.yml
-- template: use-dotnet-sdk.yml
+  - script: echo $(setTargets.targets)
+    displayName: "Print settings"
 
-- pwsh: dotnet tool install -g Microsoft.OpenApi.Hidi --prerelease
-  displayName: install hidi
+- job: publish_openapi
+  dependsOn: get_conversion_settings
+  displayName: Publish
+  strategy:
+    matrix: $[ dependencies.get_conversion_settings.outputs['setTargets.targets'] ]
+  variables:
+    targets: $[ dependencies.get_conversion_settings.outputs['setTargets.targets'] ]
 
-- pwsh: |
-    git fetch origin master
-    git switch master
-  displayName: ensure the generation happens from master latest
-  workingDirectory: $(Build.SourcesDirectory)/msgraph-metadata
+  steps:
+  # We only need the scripts
+  - checkout: self
+    displayName: checkout generator
+    fetchDepth: 1
+    persistCredentials: true
 
-- pwsh: '$(scriptsDirectory)/generate-open-api.ps1 -endpointVersion ${{ parameters.endpoint }}'
-  displayName: 'update ${{ parameters.endpoint }} open API description'
-  workingDirectory: $(Build.SourcesDirectory)/msgraph-metadata
+  - template: checkout-metadata.yml
+  - template: set-user-config.yml
+  - template: use-dotnet-sdk.yml
 
-# publish metadata as an artifact
-- task: CopyFiles@2
-  inputs:
-    sourceFolder: ${{ parameters.outputPath }}
-    contents: '**/*'
-    targetFolder: '$(Build.ArtifactStagingDirectory)'
-  displayName: Copy generated metadata
+  - pwsh: dotnet tool install -g Microsoft.OpenApi.Hidi --prerelease
+    displayName: install hidi
 
-- task: PublishBuildArtifacts@1
-  inputs:
-    pathToPublish: '$(Build.ArtifactStagingDirectory)'
-    artifactName: ${{ parameters.cleanMetadataFolder }}
+  - pwsh: |
+      git fetch origin master
+      git switch master
+    displayName: ensure the generation happens from master latest
+    workingDirectory: $(Build.SourcesDirectory)/msgraph-metadata
 
-- pwsh: |
-    ./scripts/run-openapi-validation.ps1 -repoDirectory (Get-Location).Path -version "${{ parameters.endpoint }}"
-  displayName: ensure that OpenAPI docs can be parsed
-  workingDirectory: $(Build.SourcesDirectory)/msgraph-metadata
+  - pwsh: '$(scriptsDirectory)/generate-open-api.ps1 -endpointVersion ${{ parameters.endpoint }} -settings "$(conversionSettingsDirectory)/$(File)" -platformName "$(Name)"'
```

Comment:
where is the File variable supposed to come from?

Reply:
When the pipeline iterates through the conversion-settings folder, it places the values in a variable. Then each spawned job has access to the File variable.

Reply:
are you referring to this? https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/944/files#diff-aebcf22971e4a4bda2ad75e658cf157112fa98a04cb56c4d186e8e03b94e8998R29

Reply:
thanks for clarifying!